aws/resource_aws_dx_private_virtual_interface.go

```diff
@@ -0,0 +1,194 @@
+	return &schema.Resource{
+		Create: resourceAwsDxPrivateVirtualInterfaceCreate,
+		Read:   resourceAwsDxPrivateVirtualInterfaceRead,
+		Update: resourceAwsDxPrivateVirtualInterfaceUpdate,
+		Delete: resourceAwsDxPrivateVirtualInterfaceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"connection_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vpn_gateway_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"dx_gateway_id"},
+			},
+			"dx_gateway_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"vpn_gateway_id"},
+			},
+			"vlan": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(1, 4094),
+			},
+			"bgp_asn": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"bgp_auth_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"address_family": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{directconnect.AddressFamilyIpv4, directconnect.AddressFamilyIpv6}, false),
+			},
+			"customer_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"amazon_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"tags": tagsSchema(),
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+	}
+}
+
+func resourceAwsDxPrivateVirtualInterfaceCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	vgwIdRaw, vgwOk := d.GetOk("vpn_gateway_id")
+	dxgwIdRaw, dxgwOk := d.GetOk("dx_gateway_id")
+	if vgwOk == dxgwOk {
+		return fmt.Errorf(
+			"One of ['vpn_gateway_id', 'dx_gateway_id'] must be set to create a Direct Connect private virtual interface")
+	}
+
+	req := &directconnect.CreatePrivateVirtualInterfaceInput{
+		ConnectionId: aws.String(d.Get("connection_id").(string)),
+		NewPrivateVirtualInterface: &directconnect.NewPrivateVirtualInterface{
+			VirtualInterfaceName: aws.String(d.Get("name").(string)),
+			Vlan:                 aws.Int64(int64(d.Get("vlan").(int))),
+			Asn:                  aws.Int64(int64(d.Get("bgp_asn").(int))),
+			AddressFamily:        aws.String(d.Get("address_family").(string)),
+		},
+	}
+	if vgwOk {
```

Comment:
For easier module support should this, and the other `d.GetOk()` conditionals below, also check for `!= ""`?

Reply:
👍